dotnet-interactive.Tests/HttpApiTests.cs

```diff
@@ -198,53 +202,71 @@ public async Task When_subkernel_does_not_exist_then_it_returns_404()
             response.StatusCode.Should().Be(HttpStatusCode.NotFound);
         }
 
+        [Theory]
+        [InlineData(Language.CSharp, "var x = 1234;", 0, 10, "readonly struct System.Int32")]
+        public async Task lsp_textDocument_hover_returns_expected_result(Language language, string code, int line, int character, string expected)
+        {
+            using var _ = new AssertionScope();
+            var request = new HoverParams(
+                TextDocument.FromDocumentContents(code),
+                new Lsp.Position(line, character));
+            var response = await GetServer(language).HttpClient.PostObjectAsJsonAsync("/lsp/textDocument/hover", request);
+            await response.ShouldSucceed();
+            var responseJson = await response.Content.ReadAsStringAsync();
+            var hoverResponse = LspSerializer.JsonSerializer.DeserializeFromString<HoverResponse>(responseJson);
+            hoverResponse.Contents.Kind.Should().Be(Lsp.MarkupKind.Markdown);
+            hoverResponse.Contents.Value.Should().Be(expected);
+        }
+
         [Theory]
         [InlineData(Language.CSharp)]
-        public async Task lsp_textDocument_hover_returns_expected_placeholder(Language language)
+        public async Task lsp_textDocument_hover_returns_no_result_on_malformed_request_object(Language language)
         {
             using var _ = new AssertionScope();
-            var request = JObject.Parse(@"
-{
-    ""textDocument"": {
-        ""uri"": ""path/to/document""
-    },
-    ""position"": {
-        ""line"": 1,
-        ""character"": 2
-    }
-}
-");
-            var response = await GetServer(language).HttpClient.PostJsonAsync("/lsp/textDocument/hover", request);
+            var request = new
+            {
+                SomeField = 1,
+                SomeOtherField = "test",
+            };
+            var response = await GetServer(language).HttpClient.PostObjectAsJsonAsync("/lsp/textDocument/hover", request);
             await response.ShouldSucceed();
```

Comment:
>   await response.ShouldSucceed(); [](start = 9, length = 34)

Why not return a 400?

Reply:
The regular stdio LSP protocol dictates that a value must be returned and eventually the `null` returned could (should?) be more descriptive error info that might be displayed to the user (e.g., when you F12 on a comment in C# you get a dialog "Cannot navigate to the symbol under the caret.")  It seemed more appropriate to return 200/OK along with an actionable payload, even if it's an error case.

_However_, there's no spec for LSP over HTTP, so we're blazing a trail here.  Ultimately I don't feel too strongly about it and I'll yield to you on this.

Reply:
There are two clients we're targeting right now:

1) A custom web client for use with our Jupyter shims, which we'll write. This isn't strict LSP, and we can potentially even use the Try .NET JavaScript code for this. Since a malformed request is a programming error on our part in any case, a 400 is valuable and appropriate. 

2) VS Code, which doesn't talk over HTTP.

I think following LSP strictly is useful for the second case only.

Thoughts?